ampersand-select-view.js

```diff
@@ -359,15 +359,26 @@ module.exports = View.extend({
             if (!model) throw new Error('model or model idAttribute not found in options collection');
             return this.yieldModel ? model : model[this.idAttribute];
         } else if (Array.isArray(this.options)) {
-            // find value value in options array
-            // find option, formatted [['val', 'text'], ...]
-            if (this.options.length && Array.isArray(this.options[0])) {
-                for (var i = this.options.length - 1; i >= 0; i--) {
-                    if (this.options[i][0] == value) return this.options[i];
+            var i = this.options.length - 1;
+            if (i >= 0) {
+                // find value in collection as [['val','text'], ...]
+                if (Array.isArray(this.options[0])) {
+                    for (i; i >= 0; i--) {
```

Comment:
I did this to avoid the `'i' is already defined.` error that the pre-commit hook throws. What do you think about this as a solution? Declaring i at the top of the function, and then setting it for each loop:

```
    getOptionByValue: function(value) {
        var model, i;
        if (this.options.isCollection) {
            // find value in collection, error if no model found
            if (this.options.indexOf(value) === -1) model = this.getModelForId(value);
            else model = value;
            if (!model) throw new Error('model or model idAttribute not found in options collection');
            return this.yieldModel ? model : model[this.idAttribute];
        } else if (Array.isArray(this.options)) {
            if (this.options.length > 0) {
                // find value in collection as [['val','text'], ...]
                if (Array.isArray(this.options[0])) {
                    for (i = this.options.length - 1; i >= 0; i--) {
                        if (this.options[i][0] == value) {
                            return this.options[i][0];
                        }
                    }
                // find value in collection as ['val', ...]
                } else {
                    for (i = this.options.length - 1; i >= 0; i--) {
                        if (this.options[i] == value) {
                            return this.options[i];
                        }
                    }
                }
            } else {
                throw new Error('no options provided');
            }
            throw new Error('value not in set of provided options');
        }
        throw new Error('select option set invalid');
    },
```